editoast/src/views/timetable.rs

```diff
@@ -933,4 +989,126 @@ mod tests {
             .json_into();
         assert_eq!(&response.error_type, "editoast:timetable:NotFound")
     }
+
+    #[test]
+    fn train_id_parse() {
+        let exception_key = "5062360642-f8f8-44ef-9f37-22cd0f5fcc07".to_string();
+        let index = 1;
+        let paced_train_id = 50;
+        let paced_train_occurence_id = TrainId::PacedTrainOccurrence {
+            paced_train_id,
+            index,
+        }
+        .to_string()
+        .parse::<TrainId>()
+        .unwrap();
+
+        assert_eq!(
+            paced_train_occurence_id,
+            TrainId::PacedTrainOccurrence {
+                paced_train_id,
+                index
+            }
+        );
+
+        let paced_train_exception_id = TrainId::PacedTrainException {
+            paced_train_id,
+            exception_key: exception_key.clone(),
+            index: Some(index),
+        }
+        .to_string()
+        .parse::<TrainId>()
+        .unwrap();
+        assert_eq!(
+            paced_train_exception_id,
+            TrainId::PacedTrainException {
+                paced_train_id,
+                exception_key: exception_key.clone(),
+                index: Some(1)
+            }
+        );
+
+        let paced_train_exception_id = TrainId::PacedTrainException {
+            paced_train_id,
+            exception_key: exception_key.clone(),
+            index: None,
+        }
+        .to_string()
+        .parse::<TrainId>()
+        .unwrap();
+        assert_eq!(
+            paced_train_exception_id,
+            TrainId::PacedTrainException {
+                paced_train_id,
+                exception_key: exception_key.clone(),
+                index: None
+            }
+        );
+
+        let train_schedule_id = TrainId::TrainSchedule(30)
+            .to_string()
+            .parse::<TrainId>()
+            .unwrap();
+        assert_eq!(train_schedule_id, TrainId::TrainSchedule(30));
+    }
+
+    #[test]
+    fn train_id_parse_fails() {
```

Comment:
Same remark as above, please do one test case per thing you're trying to test, it's easier to debug. In this case, you definitely should be able to use `#[rstest]` and `#[case]`.